sim/src/core/JointManager.cpp

```diff
@@ -204,18 +204,13 @@ namespace mars {
     }
 
     void JointManager::removeJointByIDs(unsigned long id1, unsigned long id2) {
-      map<unsigned long, SimJoint*>::iterator iter;
       iMutex.lock();
```

Comment:
Here we have a problem with the iMutex, this method would produce a deadlock because getIDByNodeIDs also tries to lock the iMutex. The mutex handling can be removed from this method completely.